src/output-parser.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { clouddeploy_v1 } from 'googleapis';
+import { CreateCloudDeployReleaseOutputs } from './main';
+import { errorMessage, presence } from '@google-github-actions/actions-utils';
+
+/**
+ * parseCreateReleaseResponse parses the gcloud command response for create-release
+ * into a common object
+ *
+ * @param stdout
+ * @returns DeployCloudRunOutputs
+ */
+export function parseCreateReleaseResponse(
+  stdout: string | undefined,
+): CreateCloudDeployReleaseOutputs {
+  try {
+    stdout = presence(stdout);
+    if (!stdout || stdout === '{}' || stdout === '[]') {
+      return {};
+    }
+
+    const outputJSON: clouddeploy_v1.Schema$Release = JSON.parse(stdout)[0];
+    const name = outputJSON?.name;
+
+    if (!name) {
+      return {};
+    }
+
+    const nameSplit = name.split('/');
+    const linkPrefix = 'https://console.cloud.google.com/deploy/delivery-pipelines';
+    const link = `${linkPrefix}/${nameSplit[3]}/${nameSplit[5]}/releases/${nameSplit[7]}?project=${nameSplit[1]}`;
```

Comment:
There's no guarantees that nameSplit has 8 components.

Reply:
The `name` field in the output comes from the Cloud Deploy `Release` object, so I think it should be in this defined format (see the `name` entry in the table here):
https://cloud.google.com/deploy/docs/api/reference/rest/v1/projects.locations.deliveryPipelines.releases#Release
Should I be more defensive and build in some additional validation that the returned output matches the spec?

Reply:
That assumes the API never changes. We should check the length of nameSplit to be defensive.

Reply:
Fair point -- do you think it's worth going the regular expression route to check all the fields are found and conform to the spec? Or that a simple check on the number of fields is sufficient?

Reply:
I would just check the number of elements.